chia/wallet/trade_manager.py

```diff
@@ -354,7 +354,7 @@ async def check_offer_validity(self, offer: Offer) -> bool:
             [c.name() for c in non_ephemeral_removals]
         )
         assert coin_states is not None
```

Comment:
Had this discussion with Kyle recently about using `assert`. Not sure why we have this assert - get_coin_state() as defined doesn't return Optional, but if it did, is this any different from an empty list? If this is really an exceptional case, then I would raise something. - so to me this should be either
```
 if coin_states is None:
    raise TypeError("coin_states should not be NoneType")
```
or (assuming an empty list would always be false ?)
```
if not coin_states:
  return False
```

Reply:
I think this is legacy maybe because it can never be none for a hilarious reason: https://github.com/Chia-Network/chia-blockchain/blob/55c4eb3d289bcb99f92da259ed9b1d1d5b8d589c/chia/wallet/wallet_node.py#L1342

Reply:
I can take it out

Reply:
If both sides end up having length == 0  this will return True - is that correct?